portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -31,12 +30,22 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  ArrayList<String> msglist = new ArrayList<String> (Arrays.asList("Hi!","Ni Hao!","Bonjour!","Hola!"));
-  
+  ArrayList<String> msglist = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
     response.getWriter().println(new Gson().toJson(msglist));
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String inputMsg = request.getParameter("comment-input");
+    if (!inputMsg.isEmpty()) {
```

Comment:
Seems like getParameter() actually returns a null. Checking isEmpty() is not really the same as checking not null.